pkg/virtual/workspaces/registry/rest.go

```diff
@@ -286,14 +299,29 @@ func (s *REST) authorizeForUser(ctx context.Context, orgClusterName logicalclust
 
 	}
 
+	// We need to impersonate the name of the user here, because the user Home workspace
+	// might be created on-the-fly when receiving the SAR call.
+	// And to automatically create the Home workspace, we need to be the right user.
+	// Additionally, we need to keep the `system:masters` group while impersonating,
+	// in order to have the permission to create the SAR in all cases.
+	// Note: This is not a security risk since we known that the virtual workspace user
+	// is already accessing the KCP shard as `system:masters`.
+	impersonatedClusterClient, err := s.impersonatedkubeClusterClient(&kuser.DefaultInfo{
```

Comment:
why doesn't this mean privilege escalation, i.e. if the SAR triggers a home workspace creation for a user who must not create a home workspace?